hotel-management/src/components/Toast/ToastProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import { ReactElement, useState } from 'react';
+
+import { Toast } from './Toast';
+import { ToastContext } from './ToastContext';
+import { ToastType, ToastItem } from './type';
+
+type ToastProviderProps = {
+  children: ReactElement;
+};
+
+export const ToastProvider = ({ children }: ToastProviderProps) => {
+  const [toasts, setToasts] = useState<ToastItem[] | []>([]);
+
+  const handleAddToast = (message: string, type: ToastType) => {
+    const newToast = {
+      id: crypto.randomUUID(),
+      message,
+      type,
+    };
+
+    setToasts(prev => [...prev, newToast]);
+  };
+
+  const handleRemoveToast = (id: string) => {
+    setToasts(prev => prev.filter(toast => toast.id !== id));
+  };
+
+  return (
+    <ToastContext.Provider
+      value={{ add: handleAddToast, remove: handleRemoveToast }}
+    >
+      {children}{' '}
+      <div className="toast-container">
+        {toasts &&
+          toasts.map(toast => (
+            <Toast
+              key={toast.id}
+              id={toast.id}
+              message={toast.message}
+              type={toast.type}
+              removeToast={() => handleRemoveToast(toast.id)}
```

Comment:
`removeToast` has type `(id: string) => void` so it should be `removeToast={handleRemoveToast}`